Tests/SentryTests/SentryClientTests.swift

```diff
@@ -231,6 +231,20 @@ class SentryClientTest: XCTestCase {
         assertEventNotSent(eventId: eventId)
     }
 
+    func testBeforeSendReturnsNewEvent_NewEventSent() {
+        let newEvent = Event()
+        let eventId = fixture.getSut(configureOptions: { options in
+            options.beforeSend = { _ in
+                newEvent
+            }
+        }).capture(message: message, scope: nil)
+
+        XCTAssertEqual(newEvent.eventId, eventId)
+        assertLastSentEvent { actual in
+            XCTAssertEqual(newEvent.eventId, actual.eventId)
```

Comment:
`m`: Can we also assert that the event passed to `beforeSend` is the `preparedEvent `. We can do this by checking for example that a custom release name of the options is applied to the actual event.